analytics/pubstack/eventchannel/eventchannel.go

```diff
@@ -94,9 +94,9 @@ func (c *EventChannel) flush() {
 	}
 
 	// finish writing gzip header
-	err := c.gz.Flush()
+	err := c.gz.Close()
```

Comment:
Makes sense. Looking at the gzip code, Flush empties the memory buffer but Close also writes a footer.

After you call Close though, the gz writer is left in a bad state. This is normally solved via `c.reset()`, but if there's an error calling Close the reset doesn't occur and the state is left in a potentially bad place.

I can guide on solving this, but first I need to know what you want to happen if the Close fails. Would you like to try again or dump the data, log an error, and move on?

Reply:
Right now, an error log is good.
It could be good to implement another behavior in the future :)

What do you think about deferring the reset right before the close call?

Reply:
Like that
```go
        defer c.reset()
       
        // finish writing gzip header
	err := c.gz.Close()
	if err != nil {
		glog.Warning("[pubstack] fail to close gzipped buffer")
		return
	}
```

Reply:
Cool. That works for me. The behavior of multiple `defer` is documented by the language, so `c.reset()` will be run before the lock. That's a little subtle, but safe. If you're good with it, so am I.

You may also want to consider adding a metric for failed writes, so hosts can monitor and setup alerts. That can certainly be a separate PR down the line.